### ISSUE TITLE
VMF scaling denominator was inf

### ISSUE THREAD

**Issue by rana-alshaikh** [clara-labs/spherecluster#10]
Hi
I tried to run the soft version, and this error appears 'VMF scaling denominator was inf'. what could be the reasons for that.
Another question is there any way to estimate the number of the clusters.

Thanks

**Comment by jasonlaska** [comment]
It would be great to know a bit about the data you're working with.  That exception is raised when the dimension is too large.

**Comment by ivarru** [comment]
I experience the same problem when the dimension is relatively low. For example, the test test_integration_dense in test_von_mises_fisher_mixture.py raises this exception most of the time if I let n_features = 10 (and fixes params_in to {"posterior_type": "soft"}).

**Comment by itamar-dw** [comment]
see my comment here https://github.com/jasonlaska/spherecluster/issues/17
**EDIT**
It is actually straight forward to work in log scale using the exponentially-scaled modified Bessel functions:
```
from numpy import i0e  # exp-scaled modified Bessel function of first kind order 0, I_0                                                                                                                               
from scipy.special import ive  # exp-scaled modified Bessel function of first kind, I_v   
```

**Comment by jasonlaska** [comment]
Thanks @itamar-dw would you like to add a PR for this change?

**Comment by itamar-dw** [comment]
sure. 
I tried to create a new branch and push it, but I got:

> remote: Permission to jasonlaska/spherecluster.git denied to itamar-dw

**Comment by jasonlaska** [comment]
@itamar-dw I saw one branch come up but it was derived off of master.  Could you try branching off `develop` (pull the latest develop first or rebase), then tag me on the PR?  I will merge it in after testing and reviewing.  

You should be able to submit a PR for this project via the github UI by forking to your own account first https://git-scm.com/book/en/v2/GitHub-Contributing-to-a-Project

Let me know if you need help.

**Comment by jasonlaska** [comment]
Alternatively, I'm happy to cherry-pick from a gist, if that's easier.

**Comment by itamar-dw** [comment]
Yes I created a PR then I realized I have to create a branch first. I'll do it as you suggested, that's an opportunity to learn how to do it :)
